### ISSUE TITLE
Check images for dark mode compatibility

### ISSUE THREAD

**Issue by astroDimitrios** [carpentries/instructor-training#1715]
### What is the problem?

There is an image in the memory management section that has no background (transparent) and the text is black so can't be seen in dark mode: https://carpentries.github.io/instructor-training/05-memory.html#chunking

Maybe this can have a slightly grey background to make it work with both themes?

### Location of problem (optional)

https://carpentries.github.io/instructor-training/05-memory.html#chunking

**Comment by brownsarahm** [comment]
@froggleston does sandpaper support using different images for dark and light mode?

**Comment by astroDimitrios** [comment]
@brownsarahm Not currently. When a user chooses dark mode atm any normal images are [darkened slightly and desaturated](https://github.com/carpentries/varnish/blob/1f1e219d53c8ac42d5f2864cf2ecaf3f84ff4c0c/source/stylesheets/dark.scss#L49). It might be possible to have completely different images for light and dark mode but this would require changes to sandpaper and varnish so that images are wrapped in the `<picture>` tag (and probably more javascript).

I think SVG's if inlined can contain a prefers-color-scheme media query but I haven't tested using any inline svg with the Workbench.

When [Mermaid diagrams are in Varnish](https://github.com/carpentries/varnish/pull/125) these automatically respect the colour theme.

**Comment by brownsarahm** [comment]
that mermaid note is good to know. 

I think it's worth merging Karen's PR for now and keeping this open while we make a more holistic plan. @ndporter does that make sense to you?

**Comment by ndporter** [comment]
I went ahead and merged @karenword's fix of replacing the SVG with PNG. We should definitely come back to this problem once there's a more elegant solution. Also, I didn't change the merge to remove the old SVG version (to keep it easily available/editable) but before closing this issue, any unused files should be removed.

**Comment by ndporter** [comment]
Adding the mental model [diagrams with the ball](https://carpentries.github.io/instructor-training/instructor/02-practice-learning.html#mapping-a-mental-model) to the to-do list for this one

**Comment by ha0ye** [comment]
It might also be worth checking if a bit of css for inserting the images or editing on the svg to have a background would be a viable alternative solution. see https://stackoverflow.com/questions/70174816/how-to-change-background-color-in-svg


Not sure it matters greatly, but I like the idea of svg images that scale better and are lighter-weight than png.

**Comment by brownsarahm** [comment]
i 💯 support the goal of making svgs work better. I think it's reasonable to plan that we design all of the svgs for a light background and then insert it manually for darkmode, but I could be missing use cases

**Comment by astroDimitrios** [comment]
I have opened https://github.com/carpentries/varnish/pull/153 which could use some testing to add support for dark mode images :)

**Comment by ndporter** [comment]
Circling back to this after teaching IT again because some of the images still don't work well in dark mode. @astroDimitrios are we close to ready to deploy something in the infrastructure or should we just make no-transparency versions of all the images?

**Comment by astroDimitrios** [comment]
@ndporter This PR for Varnish https://github.com/carpentries/varnish/pull/153 has a fix that I have just updated to fix conflicts @ErinBecker 

I have been testing it on the Met Office git lessons which are linked from that PR :)

**Comment by astroDimitrios** [comment]
I'm not sure if the docs have been updated yet.

If you have `image.png` in your material:

a) provide a dark mode version in the same directory as `image.png` with the name `image-dark.png`
b) don't provide one and a dark mode image filter is applied

**Comment by froggleston** [comment]
Apologies @brownsarahm , yes we need to add in instructions to sandpaper-docs - it's [on our list!](https://github.com/carpentries/sandpaper-docs/issues/220)

**Comment by ndporter** [comment]
@astroDimitrios is there a way to force the check (what you labeled (b))? I'm looking at, for example, the image of mental models in [this section](https://carpentries.github.io/instructor-training/02-practice-learning.html#the-power-and-limitations-of-analogies) of the Instructor Training. I did try refreshing the page to make sure it was rendering the newest version but I'm assuming there's something I need in the repo to trigger it to check and adapt.

**Comment by ha0ye** [comment]
@ndporter I would try triggering a new build. Based on this report from the last build, it seems it was using varnish 1.0.5, and the patch was merged in 1.0.6
https://github.com/carpentries/instructor-training/actions/runs/14848903476/job/41688665103#step:5:567

**Comment by astroDimitrios** [comment]
Yes re-building as @ha0ye said should build with the latest version of Varnish. You can check the build log when you re-trigger the build to see if the correct Varnish was used and in dark mode on the rendered website any images that don't have a dark mode version should have the `dark-filter` class.

![Image](https://github.com/user-attachments/assets/9fa1d613-8e3e-4666-bc9c-61741432ee87)
